sdk/storage/Azure.Storage.Files.DataLake/README.md

```diff
@@ -33,39 +34,178 @@ az storage account create --name MyStorageAccount --resource-group MyResourceGro
 
 ## Key concepts
 
-TODO: Add Key Concepts
+This preview package for .NET includes ADLS Gen2 specific API support made available in Blob SDK. This includes:
+1. New directory level operations (Create, Rename/Move, Delete) for hierarchical namespace enabled (HNS) storage accounts. For HNS enabled accounts, the rename/move operations are atomic.
+2. Permission related operations (Get/Set ACLs) for hierarchical namespace enabled (HNS) accounts. 
+
+HNS enabled accounts in ADLS Gen2 can also now leverage most of the operations available in Blob SDK. Support for File level semantics for ADLS Gen2 is planned to be made available in Blob SDK in a later release. In the meantime, please find below mapping for ADLS Gen2 terminology to Blob terminology.
```

Comment:
We've generally avoided talking about future plans in our readmes like this.  If there isn't a major reason we're doing so, we might want to remove this paragraph.